### PR TITLE
Add missing track entries to the logfile

### PR DESCRIPTION
**Pull request by jamesob**
It'd be really nice to be able to passively record which incomplete
albums are in your collection by grepping the logfile.

Some refactoring acrobatics were necessary to get the logger in a
place that allows this, but despite large code movements these
changes are mostly superficial.

### CODE REVIEW

File: beets/ui/commands.py [hunk 1]
```diff
@@ -821,6 +471,361 @@ def should_resume(self, path):
...
+                ))
+        for track_info in match.extra_tracks:
+            line = ' ! %s (#%s)' % (track_info.title, format_index(track_info))
+            if track_info.length:
+                line += ' (%s)' % ui.human_seconds_short(track_info.length)
+            print_(ui.colorize('text_warning', line))
+            self.logger.info("missing track: {0} from {1}".format(
+                track_info.title,
+                match.info.album,
+            ))
```
Comment (sampsyo) [inline beets/ui/commands.py]: Can you explain a little more what you're doing with this log output? Note that this prints to the global beets log, not the importer log (which is a separate mechanism, confusingly). This means that, in a typical setup, the user will see messages about each extra track twice: one on stdout, once on stderr. Is this really what you intended?

Comment (jamesob) [inline beets/ui/commands.py]: Note the refactoring that makes this function into a method, and use of `self.logger` vs. `logger`, which is set up to log into file and not the console.

Comment (sampsyo) [inline beets/ui/commands.py]: Oops! My apologies for missing that obvious detail.

The other tricky thing here is that you seem to be logging the extra tracks when the match is *displayed*, not when it is applied. So if someone considers two alternative album matches for one album they have, they'll get two different log entries for this. I think this kind of logging needs to occur after duplicate resolution, etc.

**Comment by arcresu** [comment]
Thanks for your contribution to beets! This PR has been inactive for a while and there were some outstanding review points. It seems to me that the `missing` plugin is now able to provide you with the same information you could get by grepping the log file after your changes. It accepts a library query, so you could restrict its output to items with a recent `added` date. If it's really important to log the missing information per import session, then it sounds like a new plugin might be better than refactoring beets' logging internals for this purpose.

I'll close this based on what I mentioned, but if you disagree and would like to bring this PR up to date, please feel free to re-open it.

**Pull request closed without merging**